### PR TITLE
Change join and try_join to functions

### PR DESCRIPTION
**Pull request by taiki-e**
This is part of #1215.

### CODE REVIEW

File: futures-util/src/future/join.rs [hunk 1]
```diff
@@ -68,15 +69,147 @@ macro_rules! generate {
...
     (Join4, <Fut1, Fut2, Fut3, Fut4>),
 
-    /// Future for the [`join5`](super::FutureExt::join5) method.
+    /// Future for the [`join5`] function.
     (Join5, <Fut1, Fut2, Fut3, Fut4, Fut5>),
 }
+
+/// Joins the result of two futures, waiting for them both to complete.
+///
+/// This function will return a new future which awaits both this and the
```
Comment (Nemo157) [inline futures-util/src/future/join.rs]: “a new future which awaits both futures to complete”

File: futures-util/src/future/join.rs [hunk 2]
```diff
@@ -68,15 +69,147 @@ macro_rules! generate {
...
+/// `other` future to complete. The returned future will finish with a tuple
+/// of both results.
+///
+/// Note that this function consumes the receiving future and returns a
```
Comment (Nemo157) [inline futures-util/src/future/join.rs]: “this function consumes the passed futures”

File: futures-util/src/future/join.rs [hunk 3]
```diff
@@ -68,15 +69,147 @@ macro_rules! generate {
...
+/// ```
+/// #![feature(async_await, await_macro, futures_api)]
+/// # futures::executor::block_on(async {
+/// use futures::future::{self, FutureExt};
+///
+/// let a = future::ready(1);
+/// let b = future::ready(2);
+/// let c = future::ready(3);
+/// let d = future::ready(4);
+/// let tuple = a.join4(b, c, d);
```
Comment (Nemo157) [inline futures-util/src/future/join.rs]: `future::join4`

**Comment by cramertj** [comment]
Works for me, though I'm a bit worried they'll be less discoverable. Still, no less discoverable than the macros :/

**Pull request merged**